runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -298,30 +298,43 @@ public UUID getStreamID() {
     }
 
     private <R> R TXExecuteInner(Supplier<R> txFunction, boolean isMetricsEnabled) {
-        long sleepTime = 1L;
         final long maxSleepTime = 1000L;
+        long sleepTime = 1L;
         int retries = 1;
+        int maxNumOfRetries = 20; // Approximately 10 seconds.
```

Comment:
Would it make sense to have the user decide on the number of retries ? If it is default, we can set it to 20.